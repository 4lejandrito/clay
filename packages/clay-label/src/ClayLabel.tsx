/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as React from 'react';
import classNames from 'classnames';

type DisplayType = 'secondary' | 'info' | 'warning' | 'danger' | 'success';

interface Props extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
	closeButtonProps?: React.HTMLAttributes<HTMLButtonElement>;
	displayType?: DisplayType;
	large?: boolean;
}

const ClayLabel: React.FunctionComponent<Props> = ({
	children,
	className,
	closeButtonProps,
	displayType = 'secondary',
	href,
	large = false,
	...otherProps
}) => {
	const content = (
		<>
			<span className="label-item label-item-expand">{children}</span>

			{closeButtonProps && (
				<span className="label-item label-item-after">
					<button
						{...closeButtonProps}
						className="close"
						type="button"
					>
						{closeButtonProps.children}
					</button>
				</span>
			)}
		</>
	);

	const attrs = {
		...otherProps,
		className: classNames('label', className, {
			'label-dismissible': closeButtonProps,
			'label-lg': large,
			[`label-${displayType}`]: displayType,
		}),
		href,
	};

	if (!href) {
		delete attrs.href;
	}

	return React.createElement(href ? 'a' : 'span', attrs, content);
};

export default ClayLabel;
