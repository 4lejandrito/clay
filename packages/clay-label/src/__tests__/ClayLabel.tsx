/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ClayLabel from '../ClayLabel';

describe('ClayLabel', () => {
	it('renders', () => {
		const testRenderer = TestRenderer.create(
			<ClayLabel>{'Default Label'}</ClayLabel>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});
});
