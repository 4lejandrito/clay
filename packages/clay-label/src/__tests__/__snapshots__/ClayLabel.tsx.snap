// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayLabel renders 1`] = `
<span
  className="label label-secondary"
>
  <span
    className="label-item label-item-expand"
  >
    Default Label
  </span>
</span>
`;
